src/Kurrent.Client/PersistentSubscriptions/KurrentPersistentSubscriptionsClient.Read.cs

```diff
@@ -200,30 +414,34 @@ public class PersistentSubscriptionResult : IAsyncEnumerable<ResolvedEvent>, IAs
 			public IAsyncEnumerable<PersistentSubscriptionMessage> Messages {
 				get {
 					if (Interlocked.Exchange(ref _messagesEnumerated, 1) == 1)
-                        throw new InvalidOperationException("Messages may only be enumerated once.");
+						throw new InvalidOperationException("Messages may only be enumerated once.");
 
 					return GetMessages();
 
 					async IAsyncEnumerable<PersistentSubscriptionMessage> GetMessages() {
-                        try {
-                            await foreach (var message in _channel.Reader.ReadAllAsync(_cts.Token)) {
-                                if (message is PersistentSubscriptionMessage.SubscriptionConfirmation(var subscriptionId)) 
-                                    SubscriptionId = subscriptionId;
-
-                                yield return message;
-                            }
-                        }
-                        finally {
-                            _cts.Cancel();
-                        }
+						try {
+							await foreach (var message in _channel.Reader.ReadAllAsync(_cts.Token)) {
+								if (message is PersistentSubscriptionMessage.SubscriptionConfirmation(var subscriptionId
+								    ))
```

Comment:
Can we merge this line to previous one?

Reply:
Fixed in https://github.com/EventStore/EventStore-Client-Dotnet/pull/329/commits/e8385b38d24f5eb138b8393a4acbcdc0c6416a9b